Sources/AppCoinsSDK/Domain/Entities/Product.swift

```diff
@@ -120,7 +120,7 @@ public struct Product {
                 }
             }
             
-            if let result = result { return result } else { return .failed(error: .unknown) }
+            if let result = result { return result } else { return .failed(error: .unknown(message: "Purchase failed", description: "Failed to retrive required value: result is nil at Product.swift:purchase", request: nil)) }
```

Comment:
Ortographic error: 'retrieve' instead of 'retrive'.

Reply:
Done